staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -174,6 +174,9 @@ type CustomResourceDefinitionVersion struct {
 	// Top-level and per-version columns are mutually exclusive.
 	// Per-version columns may not all be set to identical values (top-level columns should be used instead)
 	// This field is alpha-level and is only honored by servers that enable the CustomResourceWebhookConversion feature.
+	// NOTE: for CRDs created prior to 1.13 release, apiserver has defaulted the top-level additionalPrinterColumns
+	// to be non-nil. Therefore if you want to use the per-version additionalPrinterColumns, you have to use UPDATE
+	// instead of PATCH to clear the top-level additionalPrinterColumns, to avoid mutual-exclusive validation error.
```

Comment:
a patch request can set the top-level columns to null explicitly at the same time it sets the per-version columns, right? I don't think this has to dictate using UPDATE